projects/storefrontlib/shared/components/hierarchy/hierarchy-node/hierarchy-node.component.html

```diff
@@ -0,0 +1,23 @@
+<div>
+  <!-- Title Node -->
+  <ng-container *ngIf="type === 'TITLE'">
+    <cx-hierarchy-node-title
+      [options]="{
+        tree: options.tree,
+        titleReadonly: options.titleReadonly,
+        onItemRemove: options.onItemRemove
+      }">
+    </cx-hierarchy-node-title>
+  </ng-container>
+  <!-- Collapsible Node -->
+  <ng-container *ngIf="type === 'COLLAPSIBLE'">
```

Comment:
In general, it looks great! Nice job :D I still think that we can improve this component further.

So similar comment to above regarding the `node-title` component. I see that a node is either going to be collapsable (has some logic) or not-collapsable (little/zero logic). I don't think we need to differentiate these functions so much that they are separate components. 

Currently we have something like:
```
<cx-hierarchy>
  <cx-h-node-title> // Products on the go
  <cx-h-node-collapsable> // Camera component
  <cx-h-node-collapsable>  // Accessory component
```
I still think this might be a little more complicated that it needs to be. Especially given that the tree will be CMS-driven on any button press (ie. removeItem() function as to be passed in and tree is reloaded after making occ call). Also, we seem to stop nesting after the h-node. But this could be present in an example I haven't seen yet.

I think we could do something where we just use the node component and provide a class based on level to style it at different depths similar to the `cx-navigation-ui` component. For example:
```
<cx-hierarchy-node class="h-node-depth-0"> // Products on the go
  <cx-hierarchy-node class="h-node-depth-1"> // Camera component  
  <cx-hierarchy-node class="h-node-depth-1"> // Accessory component
```
This gives the possibility for the component to be more useful if we have even more depths:
```
<cx-hierarchy-node class="h-node-depth-0"> // If we want to wrap the list to make styling easier
  <cx-hierarchy-node class="h-node-depth-1"> // Products on the go
    <cx-hierarchy-node class="h-node-depth-2"> // Camera component  
    <cx-hierarchy-node class="h-node-depth-2"> // Accessory component
  <cx-hierarchy-node class="h-node-depth-1"> // Some other category
    <cx-hierarchy-node class="h-node-depth-2"> // Component  1
    <cx-hierarchy-node class="h-node-depth-2"> // Component 2
      <cx-hierarchy-node class="h-node-depth-3"> // Component 3
```

When it comes to the buttons as I mentioned before, maybe we could also pass these in the templates as they are very targeted in this instance to `bundles`. As in, not every hierarchy node with have an "edit" or "remove" button specifically. I am open to ideas on this however.

I would also encourage you to add the jsdoc-like comments to methods and properties on this component at this point. It can help with understanding the component for yourselves (and me) and identify some redundancy.

Any questions or comments about any of these points, please ask :)